Server/Jobs/CheckAndStartCIBuild.cs

```diff
@@ -53,46 +53,43 @@ public async Task Execute(long ciProjectId, long ciBuildId, CancellationToken ca
         }
 
         // Update our local repo copy and see what the wanted build config is
-        var semaphore = localTempFileLocks.GetTempFilePath($"ciRepos/{ciProjectId}", out string tempPath);
-
-        await semaphore.WaitAsync(cancellationToken);
-
-        var deserializer = new DeserializerBuilder().WithNamingConvention(CamelCaseNamingConvention.Instance)
-            .Build();
-
-        if (build.CiProject == null)
-            throw new NotLoadedModelNavigationException();
```

Comment:
@hhyyrylainen to note that this is a potential problem because the semaphore is now unreleased. This issue is fixed in any case now.

Reply:
Yeah, I see. This is exactly the type of problem to be careful with the semaphores, and in this case it seems I've been too careless. Though this exception should never be able to be thrown, if I remember right I added this to satisfy the nullability checker here, which I guess sort of proves how error prone working with semaphores directly can be.